cloudinit/config/cc_grub_dpkg.py

```diff
@@ -43,10 +44,58 @@
 import os
 
 from cloudinit import util
+from cloudinit.util import ProcessExecutionError
 
 distros = ['ubuntu', 'debian']
 
 
+def fetch_idevs(log):
+    """
+    Fetches the /dev/disk/by-id device grub is installed to.
+    Falls back to plain disk name if no by-id entry is present.
+    """
+    disk = ""
+    devices = []
+
+    try:
+        # get the root disk where the /boot directory resides.
+        disk = util.subp(['grub-probe', '-t', 'disk', '/boot'],
```

Comment:
On a GCE Debian buster image (the most convenient way I had of launch a Debian image), `grub-probe` is not in `PATH` of my normal user (because it's in `/usr/sbin`). While I suspect that Debian `cloud-init` units will have `/usr/sbin` in `PATH`, we could be safe and specify the full path here:

```suggestion
        disk = util.subp(['/usr/sbin/grub-probe', '-t', 'disk', '/boot'],
```

(We can't use the `util.which` helper to find it, because it relies on `PATH`.)

Reply:
Can we just confirm what root PATH is by default in Debian?  We consistently avoid including PATHS and expect it to include the common bin paths (including sbin).

Reply:
It is in the `PATH` of `root`, I suppose it's the problem of whoever's configuring a system if cloud-init isn't using root's `PATH`.